polygeist/test/polygeist-opt/sycl/sycl-raise-host.mlir

```diff
@@ -0,0 +1,16 @@
+// RUN: polygeist-opt --sycl-raise-host --split-input-file %s | FileCheck %s
+
+gpu.module @device_functions {
+  gpu.func @foo() kernel {
+    gpu.return
+  }
+}
+
+// CHECK-LABEL: sycl.host.kernel_name @kernel_ref -> @device_functions::@foo
+llvm.mlir.global private unnamed_addr constant @kernel_ref("foo\00") {addr_space = 0 : i32, alignment = 1 : i64, dso_local}
```

Comment:
Add a test this a global that has extern linkage/visibility to ensure the transformation does not remove the global even if it all of its references in this translation unit can be replaced wit `sycl.host.kernel_name`.